TODO

```diff
@@ -406,14 +366,11 @@ Features:
   set up the directory so that it can only be accessed if host and app are in
   order.
 
-* TPM2: add auth policy for signed PCR values to make updates easy. i.e. do
-  what tpm2_policyauthorize tool does.  To be truly useful scheme needs to be a
-  bit more elaborate though: policy probably must take some nvram based
-  generation counter into account that can only monotonically increase and can
-  be used to invalidate old PCR signatures. Otherwise people could downgrade to
-  old signed PCR sets whenever they want. Usecase: encrypt the rootfs with LUKS
-  with a key that can only be unlocked via a pristine pre-built Fedora
-  kernel+initrd.
+* TPM2: extend unlock policy to protect against version downgrades in signed
+  policies: policy probably must take some nvram based generation counter into
+  account that can only monotonically increase and can be used to invalidate
+  old PCR signatures. Otherwise people could downgrade to old signed PCR sets
+  whenever they want.
```

Comment:
I have vague memories of a specific PCR that can take a policy that stops it from ever being reset, even by the kernel, is that something that could be useful for this?

Reply:
Most PCRs cannot be reset unless you reboot. There are only a select few which can be reset by software, during system runtime. None of the PCRs are persistent, they all implicitly reset to zero on reboot.

But this is a different issue here: we want a counter that ensures that once you upgraded the OS you cannot go back with the OS and still retain access to your secrets. So the goal for that is a *persistent* counter we can increase on upgrades, and bind policy to when enrolling a disk against.